packages/fiori/src/ShellBar.ts

```diff
@@ -459,10 +165,363 @@ const metadata = {
+	_menuPopoverExpanded!: boolean;
+
+	@property({ type: Boolean, noAttribute: true })
+	_overflowPopoverExpanded!: boolean;
+
+	@property({ type: Boolean, noAttribute: true })
+	_fullWidthSearch!: boolean;
+
+	/**
+	 * Defines the <code>ui5-shellbar</code> aditional items.
+	 * <br><br>
+	 * <b>Note:</b>
+	 * You can use the &nbsp;&lt;ui5-shellbar-item>&lt;/ui5-shellbar-item>.
+	 *
+	 * @type {sap.ui.webc.fiori.IShellBarItem[]}
+	 * @name sap.ui.webc.main.ShellBar.prototype.default
+	 * @slot items
+	 * @public
+	 */
+	@slot({ type: HTMLElement, "default": true, invalidateOnChildChange: true })
+	items!: Array<ShellBarItem>;
+
+	/**
+	 * You can pass <code>ui5-avatar</code> to set the profile image/icon.
+	 * If no profile slot is set - profile will be excluded from actions.
+	 *
+	 * Note: We recommend not using the <code>size</code> attribute of <code>ui5-avatar</code> because
+	 * it should have specific size by design in the context of <code>ui5-shellbar</code> profile.
+	 * @type {sap.ui.webc.main.IAvatar}
+	 * @name sap.ui.webc.main.ShellBar.prototype.profile
+	 * @slot profile
+	 * @since 1.0.0-rc.6
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	profile!: Array<HTMLElement>;
+
+	/**
+	 * Defines the logo of the <code>ui5-shellbar</code>.
+	 * For example, you can use <code>ui5-avatar</code> or <code>img</code> elements as logo.
+	 * @type {sap.ui.webc.main.IAvatar}
+	 * @name sap.ui.webc.main.ShellBar.prototype.logo
+	 * @slot
+	 * @since 1.0.0-rc.8
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	logo!: Array<HTMLElement>;
+
+	/**
+	 * Defines the items displayed in menu after a click on the primary title.
+	 * <br><br>
+	 * <b>Note:</b>
+	 * You can use the &nbsp;&lt;ui5-li>&lt;/ui5-li> and its ancestors.
+	 *
+	 * @type {sap.ui.webc.main.IListItem[]}
+	 * @name sap.ui.webc.main.ShellBar.prototype.menuItems
+	 * @slot
+	 * @since 0.10
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	menuItems!: Array<HTMLElement>;
+
+	/**
+	 * Defines the <code>ui5-input</code>, that will be used as a search field.
+	 *
+	 * @type {sap.ui.webc.main.IInput}
+	 * name sap.ui.webc.main.ShellBar.prototype.searchField
+	 * @slot
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	searchField!: Array<Input>;
+
+	/**
+	 * Defines a <code>ui5-button</code> in the bar that will be placed in the beginning.
+	 * We encourage this slot to be used for a back or home button.
+	 * It gets overstyled to match ShellBar's styling.
+	 *
+	 * @type {sap.ui.webc.main.IButton}
+	 * name sap.ui.webc.main.ShellBar.prototype.startButton
+	 * @slot
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	startButton!: Array<Button>;
+
+	static i18nBundle: I18nBundle;
+	overflowPopover?: Popover | null;
+	menuPopover?: Popover | null;
+	_isInitialRendering: boolean;
+	_defaultItemPressPrevented: boolean;
+	menuItemsObserver: MutationObserver;
+	coPilot!: object;
+	_debounceInterval?: Timeout | null;
+	_hiddenIcons!: Array<IShelBarItemInfo>;
```

Comment:
_hiddenIcons is not initialized in the contructor and does not have initial value, so using "!:" is not correct as before onBeforeRendering this variable is actually undefined.

- metadata property with default value different from undefined - use "!:"
- metadata property/metadata with default value = undefined - use "?:"
- variable initialised in the constructor use ":"
- variable not initialised in the constructor without an initial value - use "?:"